roqoqo/src/operations/analog_operations.rs

```diff
@@ -0,0 +1,164 @@
+}
+
+#[allow(non_upper_case_globals)]
+const TAGS_ApplyConstantSpinHamiltonian: &[&str; 3] = &[
+    "Operation",
+    "SpinsAnalogOperation",
+    "ApplyConstantSpinHamiltonian",
+];
+
+impl ImplementedIn1point10 for ApplyConstantSpinHamiltonian {}
+
+impl OperateSpinsAnalog for ApplyConstantSpinHamiltonian {
+    fn spin(&self) -> Result<Vec<usize>, RoqoqoError> {
+        let mut qubit_set = HashSet::new();
+        for pps in self.hamiltonian.keys() {
+            for (index, _) in pps.iter() {
+                qubit_set.insert(*index);
+            }
+        }
+        let mut qubits = Vec::from_iter(qubit_set);
+        qubits.sort();
+        Ok(qubits)
+    }
+}
+impl SupportedVersion for ApplyConstantSpinHamiltonian {
+    fn minimum_supported_roqoqo_version(&self) -> (u32, u32, u32) {
+        (1, 10, 1)
+    }
+}
+
+impl InvolveQubits for ApplyConstantSpinHamiltonian {
+    fn involved_qubits(&self) -> InvolvedQubits {
+        InvolvedQubits::All
+    }
+}
+
+/// Implements the continuous time, time-dependent spin Hamiltonian
+#[derive(Debug, Clone, PartialEq, roqoqo_derive::Operate)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
+pub struct ApplyTimeDependentSpinHamiltonian {
+    /// Hamiltonian to be simulated.
+    hamiltonian: SpinHamiltonian,
+    /// Range of time stored as a vector. The total duration of the simulations is given by the last value in the range.
+    time: Vec<f64>,
+    /// Values of time-dependent parameters, appearing in `hamiltonian`, at instances given by the vector `time`.
+    values: HashMap<String, Vec<f64>>,
+}
+
+#[allow(non_upper_case_globals)]
+const TAGS_ApplyTimeDependentSpinHamiltonian: &[&str; 3] = &[
+    "Operation",
+    "SpinsAnalogOperation",
+    "ApplyTimeDependentSpinHamiltonian",
+];
+
+impl ImplementedIn1point10 for ApplyTimeDependentSpinHamiltonian {}
+
+impl OperateSpinsAnalog for ApplyTimeDependentSpinHamiltonian {
+    fn spin(&self) -> Result<Vec<usize>, RoqoqoError> {
+        let mut qubit_set = HashSet::new();
+        for pps in self.hamiltonian.keys() {
+            for (index, _) in pps.iter() {
+                qubit_set.insert(*index);
+            }
+        }
+        let mut qubits = Vec::from_iter(qubit_set);
+        qubits.sort();
+        Ok(qubits)
+    }
+}
+
+impl SupportedVersion for ApplyTimeDependentSpinHamiltonian {
+    fn minimum_supported_roqoqo_version(&self) -> (u32, u32, u32) {
+        (1, 10, 1)
+    }
+}
+
+impl InvolveQubits for ApplyTimeDependentSpinHamiltonian {
+    fn involved_qubits(&self) -> InvolvedQubits {
+        InvolvedQubits::All
+    }
+}
+
+impl Substitute for ApplyConstantSpinHamiltonian {
+    /// Remaps qubits in operations in clone of the operation.
+    fn remap_qubits(&self, _mapping: &HashMap<usize, usize>) -> Result<Self, RoqoqoError> {
+        Ok(ApplyConstantSpinHamiltonian::new(
+            self.hamiltonian.clone(),
```

Comment:
Should the qubits in the hamiltonian be remapped according to the `mapping` input?

Reply:
I missed this. I will have to write an unique implementation for this.